tests/test_manager.py

```diff
@@ -221,3 +220,18 @@ async def test_lifespan_not_supported(app: typing.Callable) -> None:
     with pytest.raises(LifespanNotSupported):
         async with LifespanManager(app):
             pass  # pragma: no cover
+
+
+@pytest.mark.usefixtures("concurrency")
+async def test_lifespan_state_async_cm() -> None:
+    async def app(scope: Scope, receive: Receive, send: Send) -> None:
```

Comment:
It seems `.state` is public and `__aenter__()` is changed to return the manager itself just so we can run this test.

Instead I think we should inspect `scope["state"]` inside the app.